src/main/java/unicash/logic/parser/FilterCommandParser.java

```diff
@@ -0,0 +1,160 @@
+package unicash.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static unicash.logic.UniCashMessages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static unicash.logic.parser.CliSyntax.PREFIX_AMOUNT;
+import static unicash.logic.parser.CliSyntax.PREFIX_CATEGORY;
+import static unicash.logic.parser.CliSyntax.PREFIX_DATETIME;
+import static unicash.logic.parser.CliSyntax.PREFIX_LOCATION;
+import static unicash.logic.parser.CliSyntax.PREFIX_NAME;
+import static unicash.logic.parser.CliSyntax.PREFIX_TYPE;
+
+import java.util.List;
+
+import unicash.commons.util.ToStringBuilder;
+import unicash.logic.commands.FilterCommand;
+import unicash.logic.parser.exceptions.ParseException;
+import unicash.model.category.Category;
+import unicash.model.commons.Amount;
+import unicash.model.transaction.DateTime;
+import unicash.model.transaction.Location;
+import unicash.model.transaction.Name;
+import unicash.model.transaction.Type;
+import unicash.model.transaction.predicates.TransactionContainsAllKeywordsPredicate;
+
+
+/**
+ * Parses input arguments and creates a new FilterCommand object
+ */
+public class FilterCommandParser implements Parser<FilterCommand> {
+
+    private TransactionContainsAllKeywordsPredicate filterPredicate;
+
+    /**
+     * Creates a {@code FilterCommandParser} object with a default
+     * {@code TransactionContainsAllKeywordsPredicate} predicate.
+     */
+    FilterCommandParser() {
+        this.filterPredicate = new TransactionContainsAllKeywordsPredicate();
```

Comment:
Can omit use of `this`

Reply:
Also I realised that you could also just initialise the `filterPredicate` where it is declared and skip the declaration of the constructor entirely

Reply:
Correct me if I'm wrong, but I believe explicit constructor declaration and readability is more in alignment with defensive programming principles. I'm not sure where I read that in last week's content, but I rmb seeing something about this somewhere